server/controllers/EventController.js

```diff
@@ -16,6 +15,64 @@ const getEvents = async (req, res, next) => {
 
   return res.json(events)
 }
+/*
+* Returns an array of JSON objects of a given length (10 if undefined), filled firs by all events that still need submitting, with the rest
+* of the array being filled with the most recent events ordered by date.
+* */
+const getEventDashboardList = async (req, res, next) => {
+
+  // Check if an array length number has been defined and store that value, otherwise default the value to 10
+  let listLength = (req.query.listLength !== undefined) ? req.query.listLength : 10
+
+  // Find & store all events from the collection
+  let allEvents = await getEventsFromDb()
+
+  // Find & store all events that still need submitting
+  let noRecordEvents = allEvents.filter((event) => {
+    return !event.hasAttendanceRecords
+  })
+
+  // Find & store all events that have allready been submitted
+  let recordEvents = allEvents.filter((event) => {
+    return event.hasAttendanceRecords
+  })
+
+  let sortedRecordEvents = recordEvents.sort((a, b) => {
+    const dateA = new Date(a.date)
+    const dateB = new Date(b.date)
+    return (dateA > dateB) ? -1 : 1
+  })
+
+  // Find how many recent events will fit into the final array, and reduce the array to only contain those objects
+  if (noRecordEvents.length >= listLength) {
+    recordEvents = []
+  } else if (recordEvents.length > listLength - noRecordEvents.length) {
+    recordEvents = recordEvents.slice(0, listLength - noRecordEvents.length)
+  }
```

Comment:
@Ray-F We should talk about this.

Reply:
@Ray-F this slice function will shorten the entire combinedEvents array down to listLength correct? However this goes against the functionality I'm trying to implement, in which the list will show all the noRecordEvents, even if it makes the array longer than listLength. That's why I've only sliced the recordEvents when there is room to put them in, otherwise we don't display any of them. Difficult to explain, but hopefully you understand what I mean.

Reply:
If this is the wanted implementation, you could also try:

`let combinedEvents = [...noRecordEvents, ...recordEvents].slice(0, Math.max(listLength, noRecordEvents.length))`

I just think the if statement along with the subtraction operators create lots of room for misinterpretation.